### PR TITLE
contrib/gorm.io/gorm.v1: add context example

### PR DESCRIPTION
**Pull request by mackjmr**
This PR adds an example of passing context when making calls through gorm.DB. This allows the query spans to inherit their parent from context. 

This example was requested in APMS-6867.

### CODE REVIEW

File: contrib/gorm.io/gorm.v1/example_test.go [hunk 1]
```diff
@@ -29,11 +37,35 @@ func ExampleOpen() {
...
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	db, err := gormtrace.Open(postgres.New(postgres.Config{Conn: sqlDb}), &gorm.Config{})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	user := User{}
```
Comment (gbbr) [inline contrib/gorm.io/gorm.v1/example_test.go:59]:
```suggestion
	var user User
```

File: contrib/gorm.io/gorm.v1/example_test.go [hunk 2]
```diff
@@ -29,11 +37,35 @@ func ExampleOpen() {
...
+	sqlDb, err := sqltrace.Open("pgx", "postgres://pqgotest:password@localhost/pqgotest?sslmode=disable")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	db, err := gormtrace.Open(postgres.New(postgres.Config{Conn: sqlDb}), &gorm.Config{})
+	if err != nil {
+		log.Fatal(err)
+	}
+
```
Comment (gbbr) [inline contrib/gorm.io/gorm.v1/example_test.go:58]: Please remove the empty lines. See https://github.com/DataDog/dd-trace-go/wiki/Style-guidelines#general-guidelines (point 3)

File: contrib/gorm.io/gorm.v1/example_test.go [hunk 3]
```diff
@@ -6,34 +6,62 @@
...
-		gorm.Model
-		Name string
-	}{}
+	var user User
 
 	// All calls through gorm.DB are now traced.
 	db.Where("name = ?", "jinzhu").First(&user)
 }
+
+func Example_context() {
```
Comment (ajgajg1134) [inline contrib/gorm.io/gorm.v1/example_test.go:44]:
```suggestion
func ExampleContext() {
```

**Review by ajgajg1134 — APPROVED** [review]
Looks good overall! Just one small nit regarding the new function name not being the same format as the other example

Thanks!! 😃

**Review by ajgajg1134 — APPROVED** [review]

**Pull request merged**